ElectricCar.cpp

```diff
@@ -1,16 +1,34 @@
 #include "ElectricCar.hpp"
 #include <iostream>
 
-ElectricCar::ElectricCar(ElectricEngine* engine)
-    : engine_(engine)
+ElectricCar::ElectricCar(ElectricEngine *engine)
+	: electricEngine_(engine)
 {
-    std::cout << __FUNCTION__ << std::endl;
+	std::cout << __FUNCTION__ << std::endl;
+}
+ElectricCar::~ElectricCar()
+{
+	delete electricEngine_;
+	std::cout << __FUNCTION__ << std::endl;
 }
 
-ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
+void ElectricCar::addEnergy() { charge(); }
+void ElectricCar::charge() { std::cout << __FUNCTION__ << std::endl; }
+ElectricEngine *ElectricCar::changeEngine(int pow, int bat)
+{
+	if (speed_ == 0)
```

Comment:
tutaj sprawdzacie speed_ a funkcja "accelerate" ustawia velocity
co to rozjechalo sie